astropy/wcs/wcsapi/wrappers/compound_wcs.py

```diff
@@ -0,0 +1,124 @@
+from functools import reduce
+
+import numpy as np
+
+from astropy.wcs.wcsapi import BaseLowLevelWCS, wcs_info_str
+
+__all__ = ['CompoundLowLevelWCS']
+
+
+def tuplesum(lists):
+    return reduce(tuple.__add__, map(tuple, lists))
+
+
+class CompoundLowLevelWCS(BaseLowLevelWCS):
+    """
+    A wrapper that takes multiple low level WCS objects and makes a compound
+    WCS that combines them.
+
+    Parameters
+    ----------
+    *wcs : `~astropy.wcs.wcsapi.BaseLowLevelWCS`
+        The WCSes to combine
+    """
+    def __init__(self, *wcs):
+        self._wcs = wcs
+
+    @property
+    def pixel_n_dim(self):
+        return sum([w.pixel_n_dim for w in self._wcs])
+
+    @property
+    def world_n_dim(self):
+        return sum([w.pixel_n_dim for w in self._wcs])
+
+    @property
+    def world_axis_physical_types(self):
+        return tuplesum([w.world_axis_physical_types for w in self._wcs])
+
+    @property
+    def world_axis_units(self):
+        return tuplesum([w.world_axis_units for w in self._wcs])
+
+    def pixel_to_world_values(self, *pixel_arrays):
+        world_arrays = []
+        for w in self._wcs:
+            pixel_arrays_sub = pixel_arrays[:w.pixel_n_dim]
+            pixel_arrays = pixel_arrays[w.pixel_n_dim:]
+            world_arrays_sub = w.pixel_to_world_values(*pixel_arrays_sub)
+            if w.world_n_dim > 1:
+                world_arrays.extend(world_arrays_sub)
+            else:
+                world_arrays.append(world_arrays_sub)
+        return tuple(world_arrays)
+
+    def world_to_pixel_values(self, *world_arrays):
+        pixel_arrays = []
+        for w in self._wcs:
+            world_arrays_sub = world_arrays[:w.world_n_dim]
+            world_arrays = world_arrays[w.world_n_dim:]
+            pixel_arrays_sub = w.world_to_pixel_values(*world_arrays_sub)
+            if w.pixel_n_dim > 1:
+                pixel_arrays.extend(pixel_arrays_sub)
+            else:
+                pixel_arrays.append(pixel_arrays_sub)
+        return tuple(pixel_arrays)
+
+    @property
+    def world_axis_object_components(self):
+        all_components = []
+        for iw, w in enumerate(self._wcs):
+            for component in w.world_axis_object_components:
+                all_components.append((f'{component[0]}_{iw}',) + component[1:])
+        return all_components
+
+    @property
+    def world_axis_object_classes(self):
+        # TODO: deal with name conflicts
```

Comment:
Should we at least detect this in the `__init__` and throw a warning if we aren't going to fix it in the first release?

Reply:
Having played with this, is this comment even valid anymore?